### ISSUE TITLE
redundant_discardable_let should not produce warning when used with async

### ISSUE THREAD

**Issue by mhosna** [realm/SwiftLint#4142]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

let can not be removed from "async let _ = await foo()" because removing it produces invalid swift syntax

```bash
$ swiftlint lint
```
Produces:

```bash
Redundant Discardable Let Violation: Prefer `_ = foo()` over `let _ = foo()` when discarding a result from a function. (redundant_discardable_let)
```

### Environment

* SwiftLint version 0.48.0

Example for RedundantDiscardableLetRule test case:
```swift
Example("async let _ = await foo()")
```

Fix:
```swift
return file.match(pattern: "(?<!async\\s)let\\s+_\\b", with: [.keyword, .keyword]).filter { range in
```

**Comment by mhosna** [comment]
https://github.com/realm/SwiftLint/pull/4143

some tasks ended with error:
ERROR: Failed to query remote execution capabilities: UNAUTHENTICATED: no authentication metadata found
not sure what that mean..

**Comment by jpsim** [comment]
@mhosna thanks for the report & fix, the unauthenticated error was unrelated to your work, I was making some changes to our CI infrastructure and things were temporarily broken.

**Comment by mhosna** [comment]
@jpsim thank you for the merge!
